### PR TITLE
feat: allow configuration of intervals

### LINKED ISSUE #359: [FEATURE REQUEST] forwarder.interval support dynamic adjustment
https://github.com/hibiken/asynq/blob/9f2c321e9804901f06b58e82f9e0ce4125d80b74/server.go#L369

### PR DESCRIPTION
**Pull request by mdibaiee**
fixes #359

### CODE REVIEW

**Review by hibiken — CHANGES_REQUESTED** [review]
Thank you for working on this!

I think (at least for now) we only need to allow configuration of "forwarder" interval through public API. 

Also, since the name "forwarder" is very much an internal detail of the package, I think we should come up with a more descriptive name.

### Background context
The forwarder goroutine is running indefinitely and periodically checking both the *scheduled* and *retry* tasks and forwarding them to *pending* state if they are ready to be processed. The interval is currently hard-coded to 5s but some users may want to customize it. Even though we allow configuration by the user, we may want to restrict it (e.g. minimum of 1s) since we don't want to be checking this every 10ms for example.

We should also document this clearly so that it shows up in the godoc, and it's clear to the user what this configuration does.

**Comment by mdibaiee** [comment]
Hey @hibiken 
Thanks for the feedback.

I worked on multiple ones since I saw your comment on the [issue](https://github.com/hibiken/asynq/issues/359):
> There're a few intervals like this one that we can make configurable. I'll try to include this change in the next release 👍

Is it a bad idea to add them all as configurable?

Regarding naming, I think it comes down to whether we make forwarder meaningful or not: we could just add the description you added as part of the documentation of the parameter and people will understand it. I find it hard to come up with a name that's self-explanatory for this case, while still being short enough. What do you think? We can do the same for all other interval configurations: explaing what exactly is the process that runs on that interval.

**Comment by hibiken** [comment]
I probably made that comment without thinking too much on it :)

I think we should be conservative when making public API changes, once you add a configuration option it's hard to remove it (e.g. requires major version update after the v1 release). I don't see much use for other intervals tweaked by the users, right now.

As for the name, I was thinking in the lines of `DelayedTaskCheckInterval` (or something simliar), since both the *scheduled* and *retry* tasks are delayed tasks (i.e. tasks that will be processed in the future). The name forwarder is something I'm not super happy with so I'd rather not expose it  :)

Let me know if you have feedback/thoughts!

**Comment by mdibaiee** [comment]
@hibiken okay, that makes sense 👍

File: server.go [hunk 1]
```diff
@@ -144,6 +144,12 @@ type Config struct {
 	//
 	// If unset or zero, the interval is set to 15 seconds.
 	HealthCheckInterval time.Duration
+
+	// DelayedTaskCheckInterval specifies the interval between checks run on scheduled and 'retry'
```
Comment (hibiken) [inline server.go:148]: nit: quote the word "scheduled" to be consistent.

File: server.go [hunk 2]
```diff
@@ -349,24 +365,28 @@ func NewServer(r RedisConnOpt, cfg Config) *Server {
 	syncer := newSyncer(syncerParams{
 		logger:     logger,
 		requestsCh: syncCh,
-		interval:   5 * time.Second,
+		interval:   defaultSyncInterval,
```
Comment (hibiken) [inline server.go:368]: revert this change.

File: server.go [hunk 3]
```diff
@@ -349,24 +365,28 @@ func NewServer(r RedisConnOpt, cfg Config) *Server {
...
 	})
 	heartbeater := newHeartbeater(heartbeaterParams{
 		logger:         logger,
 		broker:         rdb,
-		interval:       5 * time.Second,
+		interval:       defaultHeartbeatInterval,
```
Comment (hibiken) [inline server.go:373]: revert this change.

File: server.go [hunk 4]
```diff
@@ -394,7 +414,7 @@ func NewServer(r RedisConnOpt, cfg Config) *Server {
 		retryDelayFunc: delayFunc,
 		isFailureFunc:  isFailureFunc,
 		queues:         qnames,
-		interval:       1 * time.Minute,
+		interval:       defaultRecovererPollInterval,
```
Comment (hibiken) [inline server.go:417]: revert this change.

File: server.go [hunk 5]
```diff
@@ -406,7 +426,7 @@ func NewServer(r RedisConnOpt, cfg Config) *Server {
 		logger:   logger,
 		broker:   rdb,
 		queues:   qnames,
-		interval: 8 * time.Second,
+		interval: defaultJanitorInterval,
```
Comment (hibiken) [inline server.go:429]: revert this change.

**Review by hibiken — COMMENTED** [review]
Thanks for making the change quickly, I'll approve once all outstanding comments have been resolved 👍
